### DISCUSSION TITLE
Use of a TestFlight app in appium project

### DISCUSSION THREAD

**Discussion by adelZilberman0** [appium/appium#19383] [Q&A]
Hello everyone, I'm having an issue with creating the right iOS app file for my appium project.
We have the app generated in TestFlight, so I tried to use it in the project but it didn't work. 
So I was wondering if anyone succeeded using a TestFlight app in appium or if not,
what do you find the best way to create an iOS app to work with appium?

**Comment by portyu9** [comment]
Hi @adelZilberman0 

If the TestFlight build is already installed on the real device, you don't need to extract or provide its `.ipa`.

Start the pre-installed application by its bundle ID:

```json
{
  "platformName": "iOS",
  "appium:automationName": "XCUITest",
  "appium:udid": "<device-udid>",
  "appium:bundleId": "com.yourcompany.yourapp",
  "appium:noReset": true
}
```

Appium's XCUITest driver explicitly supports automating a pre-installed real-device app this way.

If you need Appium to **install the application itself**, get a properly development/ad-hoc signed `.ipa` from your Xcode/CI build instead of trying to extract the TestFlight-distributed build:

```json
"appium:app": "/path/to/your-app.ipa"
```

So the simplest fix is: **install through TestFlight manually once, then automate it using `bundleId`**, or produce a separately signed `.ipa` specifically for automation.

Please mark this answer as accepted if it helped, thank you.
